### ISSUE TITLE
Drupal account creation fails after upgrading CiviCRM when using version 1.10

### ISSUE THREAD

**Issue by lkuttner** [eileenmcnaughton/nz.co.fuzion.csvimport#62]
After upgrading to CiviCRM 5.59.4 or later, if Version 1.10 is installed, creating a new user results in a no-name CiviCRM record with a phantom Drupal account. Upgrading extension to version 1.12 resolves this issue. Posting this to help others who encounter this. This was with D7.

 When a new user is created, a white screen results and the Drupal log displays this error:
`Message	PDOException: SQLSTATE[23000]: Integrity constraint violation: 1048 Column 'entity_id' cannot be null: INSERT INTO {field_data_message_subscribe_email} (entity_type, entity_id, revision_id, bundle, delta, language, message_subscribe_email_value) VALUES (:db_insert_placeholder_0, :db_insert_placeholder_1, :db_insert_placeholder_2, :db_insert_placeholder_3, :db_insert_placeholder_4, :db_insert_placeholder_5, :db_insert_placeholder_6); Array ( [:db_insert_placeholder_0] => user [:db_insert_placeholder_1] => [:db_insert_placeholder_2] => [:db_insert_placeholder_3] => user [:db_insert_placeholder_4] => 0 [:db_insert_placeholder_5] => und [:db_insert_placeholder_6] => 1 ) in field_sql_storage_field_storage_write() (line 622 of .../htdocs/modules/field/modules/field_sql_storage/field_sql_storage.module).`

I suspect that a factor may be that cache clearing was not completing after the CiviCRM upgrade even though Drush stated otherwise.

**Comment by eileenmcnaughton** [comment]
Thanks - it seems odd that this extension woudl cause this - but I'll leave open for  a couple of months in case others hit the same

**Comment by lkuttner** [comment]
I think the Drupal issue was a side effect.  My guess is the extension was causing cache clearing to be impaired and that was in turn causing  the account creation issue.  At one point when I tried to clear the cache from the CiviCRM page on my dev server I got this debug message that made me suspect the extension.  
`Fatal error: Declaration of CRM_Csvimport_Import_Form_DataSource::preProcess() must be compatible with CRM_Import_Form_DataSource::preProcess(): void in .../htdocs/sites/all/extensions/nz.co.fuzion.csvimport/CRM/Csvimport/Import/Form/DataSource.php on line 39`
